modules/async-icq/keeper/params.go

```diff
@@ -6,27 +6,37 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// IsHostEnabled retrieves the host enabled boolean from the paramstore.
+// IsHostEnabled retrieves the host enabled boolean from the params.
 // True is returned if the host is enabled.
 func (k Keeper) IsHostEnabled(ctx sdk.Context) bool {
-	var res bool
-	k.paramSpace.Get(ctx, types.KeyHostEnabled, &res)
-	return res
+	return k.GetParams(ctx).HostEnabled
 }
 
-// GetAllowQueries retrieves the host enabled query paths from the paramstore
+// GetAllowQueries retrieves the host enabled query paths from the params
 func (k Keeper) GetAllowQueries(ctx sdk.Context) []string {
-	var res []string
-	k.paramSpace.Get(ctx, types.KeyAllowQueries, &res)
-	return res
+	return k.GetParams(ctx).AllowQueries
 }
 
-// GetParams returns the total set of the parameters.
-func (k Keeper) GetParams(ctx sdk.Context) types.Params {
-	return types.NewParams(k.IsHostEnabled(ctx), k.GetAllowQueries(ctx))
+// SetParams sets the module parameters.
+func (k Keeper) SetParams(ctx sdk.Context, p types.Params) error {
+	if err := p.Validate(); err != nil {
+		return err
+	}
+
+	store := ctx.KVStore(k.storeKey)
+	bz := k.cdc.MustMarshal(&p)
+	store.Set(types.ParamsKey, bz)
+	return nil
 }
 
-// SetParams sets the total set of the parameters.
-func (k Keeper) SetParams(ctx sdk.Context, params types.Params) {
-	k.paramSpace.SetParamSet(ctx, &params)
+// GetParams returns the current module parameters.
+func (k Keeper) GetParams(ctx sdk.Context) (p types.Params) {
+	store := ctx.KVStore(k.storeKey)
+	bz := store.Get(types.ParamsKey)
+	if bz == nil {
+		return p
```

Comment:
nit: but because named return values are being utilized here we can make use of "naked returns" by simply using the return keyword vs. return p